Add support for setting amperage limit via Number entity

conorbranagan opened the pull request:
# Changes

Adds support for setting the amperage limit on the ChargePoint charger using a `Select`. The charger must be plugged in for the new amperage to be reflected (and this appears to be the same behavior when it's set in the app).

The entity is provided the list of available values from the ChargePoint device.

# Testing

Used the provided docker-compose file to test this locally and validate it shows up as a new entity.

<img width="532" alt="Screenshot 2024-02-25 at 9 24 13 PM" src="https://github.com/mbillow/ha-chargepoint/assets/472446/3ecf1eed-0984-4116-97f7-907d9d19e237">

mbillow reviewed (changes requested):
Overally, this looks fine. I think we should be using a `select` entity instead of a `number` entity since ChargePoint is giving us a list of known good values. We are assuming that all chargers support single amp settings and not something like every 10 amps levels.

conorbranagan commented:
Thanks for looking at this and sorry for the long delay on getting back. I've updated it to use a select entity and moved this PR out of draft. Any remaining review is welcomed!

mbillow reviewed (approved):
Awesome, thanks for swapping this to a select entity. I’ll merge and release this now.

Thanks again for the contribution!
